### PR TITLE
Fix infinite "uploading image" loading on product creation for Brave users

### PR DESCRIPTION
**Pull request by Dom-Mac**
**Environment**
_Device and OS_: All
_Browser_: Mainly on Brave (Shield enabled) and Firefox, other browsers with fingerprinting protection enabled
_Reproducibility rate_: 100%

**Step to reproduce**
_On Brave_: Activate shield > Make sure that fingerprinting is blocked > Upload an image during product creation or slicer metadata updating

**Expected result**
Product created correctly, slicer's image edited correctly.

**Actual result**
_On product creation_: Infinite loading on image uploading
_On slicer's image edit_: General error "Something went wrong"

**Description**
Brave users with Shield enabled, and in general users with Fingerprinting protection enabled, experienced an infinite loading on image upload, due to:
- Blurred image creation 
- Huge sized image optimization

The error is trowhed by Pica, a peer dependency of "image-blob-reduce": "^4.1.0", a lybrary used to reduce images.
"Pica: cannot use getImageData on canvas, make sure fingerprinting protection isn't enabled"

**Solution**
This pr is for a temporary solution that mitigates the problem. 
In the future the problem will be automatically solved with an external service in care of the images handling.

### CODE REVIEW

**Comment by jjranalli** [comment]
@Dom-Mac Made a few edits:

- refactored logic in an utils (easier to cleanup when not needed anymore)
- added `cross: true` when opening modalView, so that user can close it
- Small edit to modal text
- Edited comment specifying the part that is Image compression (note: Gif compression is not handled)

Note: 
- the `includes` condition in `openFingerprintingModal` had a second string as argument, but the function expected a number (position). Realised this when added type check. I made it so that it only needs to include `Pica` but I'm not able to reproduce the issue locally so not able to check if the modal actually gets opened. Please lmk if it works and I'll merge this

File: lib/handleCreateProduct.ts [hunk 1]
```diff
@@ -75,6 +75,10 @@ export const beforeCreate = async (
       metadata["image"] = image
     }
   }
+  // TODO: remove
+  // throw new Error(
+  //   "Pica: cannot use getImageData on canvas, make sure fingerprinting protection isn't enabled"
+  // )
```
Comment (Dom-Mac) [inline lib/handleCreateProduct.ts]: I had to throw the error manually as it is not reproducible in locale, I don't know why..

**Comment by Dom-Mac** [comment]
> @Dom-Mac Made a few edits:
> 
> * refactored logic in an utils (easier to cleanup when not needed anymore)
> * added `cross: true` when opening modalView, so that user can close it
> * Small edit to modal text
> * Edited comment specifying the part that is Image compression (note: Gif compression is not handled)
> 
> Note:
> 
> * the `includes` condition in `openFingerprintingModal` had a second string as argument, but the function expected a number (position). Realised this when added type check. I made it so that it only needs to include `Pica` but I'm not able to reproduce the issue locally so not able to check if the modal actually gets opened. Please lmk if it works and I'll merge this

I didn't know how to close that modal view, nice!  👍 
About GIF compression I was 100% sure that the if confition was fileExt == "GIF" instead of fileExt != "GIF". Thanks to catch it.

About the includes method you are right, in js it takes just one string. I added a peace of code manually to throw the error as it was not reproducible in locale. It works with just "Pica", but maybe could be better to leave just "fingerprinting".

File: lib/handleCreateProduct.ts (hunk 1 above)
Comment (Dom-Mac) [inline lib/handleCreateProduct.ts]: @jjranalli this is the manually throwed error. I deleted it in the next commit.

**Comment by jjranalli** [comment]
Perfect, we can merge after you push the last commits ✌️

**Pull request merged**